includes/class-wp-smime.php

```diff
@@ -110,7 +110,19 @@ public static function encrypt ( $message, $headers, $certificates ) {
         $infile  = tempnam( sys_get_temp_dir(), 'wp_email_' );
         $outfile = $infile . '.enc';
 
-        $plaintext  = ( is_array( $headers ) ) ? implode( "\n", $headers ) : $headers;
+        $plaintext = ( is_array( $headers ) ) ? implode( "\n", $headers ) : $headers;
+
+        // add content type and UTF-8 charset if not otherwise specified or missing
+        if ( false === stripos( $plaintext, "Content-Type" ) ) {
+            $plaintext .= "Content-Type: text/plain; charset=utf-8;";
+
+        } elseif ( false === stripos( $plaintext, "charset" ) ) {
+            $plaintext = preg_replace(
+                "/(?(?!;)(Content-Type:.*?);?)(\r\n|\r|\n)*?$/im",
```

Comment:
Can you talk me through this regex? On closer inspection, it looks like you're merely appending `charset=utf-8;` to a `Content-Type` header if that string is not already present. If so, isn't there a more human-readable way of writing that than a complex regex? If not, what am I missing in your regex that is also happening here?

Reply:
You are correct, it just appends the charset. It's basically a conditional regex trying to avoid duplicate semicolons and handle different linebreaks. (https://regex101.com/r/0T63og/1)
I figured it's not that complicated, but i'll try to refactor this part.